toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1355,23 +1445,33 @@ func (g *PkgGraph) fixPrebuiltSRPMsCycle(trimmedCycle []*PkgNode) (err error) {
 
 			logger.Log.Debugf("Adding a 'PreBuilt' node '%s' with id %d.", preBuiltNode.FriendlyName(), preBuiltNode.ID())
 
-			parentNodes := g.To(currentNode.ID())
-			for parentNodes.Next() {
-				parentNode := parentNodes.Node().(*PkgNode)
-				if parentNode.Type == TypeLocalBuild && parentNode.SrpmPath == previousNode.SrpmPath {
-					g.RemoveEdge(parentNode.ID(), currentNode.ID())
-
-					err = g.AddEdge(parentNode, preBuiltNode)
-					if err != nil {
-						logger.Log.Errorf("Adding edge failed for %v -> %v", parentNode, preBuiltNode)
-						return
-					}
-				}
-			}
+			g.replaceCurrentNodeWithPrebuiltNode(currentNode, preBuiltNode, previousNode)
+			logger.Log.Infof("Cycle fixed using prebuilt node: %s with id %d", preBuiltNode.FriendlyName(), preBuiltNode.ID())
 
 			return
 		}
-
+		if resolveCyclesFromUpstream {
+			isAvailableUpstream := false
+			var clonerErr error
+			isAvailableUpstream, clonerErr = isSRPMAvailableUpstream(g, nil, currentNode.SrpmPath, ignoreVersionToResolveSelfDep, cloner)
+			/*clonerErr is not fatal. gives error if rpm is not found*/
+			if clonerErr != nil {
+				logger.Log.Debugf("Error while checking if SRPM is available upstream: %v", clonerErr)
+			}
```

Comment:
I need to modify the cloner so it doesn't return errors when `WhatProvides` doesn't find any RPMs... But for now, please consider these changes:

```suggestion
			isAvailableUpstream, clonerErr := isSRPMAvailableUpstream(g, nil, currentNode.SrpmPath, ignoreVersionToResolveSelfDep, cloner)
			// Cloner errors are not fatal - may return one when the RPM is not found.
			if clonerErr != nil {
				logger.Log.Debugf("Error while checking if SRPM is available upstream: %v", clonerErr)
				continue
			}
```